packages/hoppscotch-common/src/components/http/authorization/JWT.vue

```diff
@@ -0,0 +1,320 @@
+        :envs="envs"
+      />
+    </div>
+  </div>
+
+  <div class="ml-4 py-2 border-b border-dividerLight">
+    <label class="text-secondaryLight">
+      {{ t("authorization.jwt.headers") }}
+    </label>
+    <div ref="headersEditor" class="mt-2 h-32"></div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { useCodemirror } from "@composables/codemirror"
+import { useI18n } from "@composables/i18n"
+import { HoppRESTAuthJWT } from "@hoppscotch/data"
+import { useVModel } from "@vueuse/core"
+import { computed, reactive, ref } from "vue"
+import { AggregateEnvironment } from "~/newstore/environments"
+import IconCircle from "~icons/lucide/circle"
+import IconCircleDot from "~icons/lucide/circle-dot"
+
+const t = useI18n()
+const authTippyActions = ref<any | null>(null)
+
+const props = withDefaults(
+  defineProps<{
+    modelValue: HoppRESTAuthJWT
+    envs?: AggregateEnvironment[]
+  }>(),
+  {
+    envs: undefined,
+  }
+)
+
+const emit = defineEmits<{
+  (e: "update:modelValue", value: HoppRESTAuthJWT): void
+}>()
+
+const auth = useVModel(props, "modelValue", emit)
+
+// Template refs for CodeMirror editors
+const payloadEditor = ref<any | null>(null)
+const headersEditor = ref<any | null>(null)
+const privateKeyEditor = ref<any | null>(null)
+
+const payload = computed({
+  get: () => auth.value.payload,
+  set: (value) => {
+    auth.value = {
+      ...auth.value,
+      payload: value,
+    }
+  },
+})
+
+const jwtHeaders = computed({
+  get: () => auth.value.jwtHeaders,
+  set: (value) => {
+    auth.value = {
+      ...auth.value,
+      jwtHeaders: value,
+    }
+  },
+})
+
+const privateKey = computed({
+  get: () => auth.value.privateKey,
+  set: (value) => {
+    auth.value = {
+      ...auth.value,
+      privateKey: value,
+    }
+  },
+})
+
+// Initialize CodeMirror for payload editor
+useCodemirror(
+  payloadEditor,
+  payload,
+  reactive({
+    extendedEditorConfig: {
+      mode: "application/json",
+      readOnly: false,
+      lineWrapping: true,
+    },
+    linter: null,
+    completer: null,
+    environmentHighlights: true,
+  })
+)
+
+// Initialize CodeMirror for headers editor
+useCodemirror(
+  headersEditor,
+  jwtHeaders,
+  reactive({
+    extendedEditorConfig: {
+      mode: "application/json",
+      readOnly: false,
+      lineWrapping: true,
+    },
+    linter: null,
+    completer: null,
+    environmentHighlights: true,
+  })
+)
+
+useCodemirror(
+  privateKeyEditor,
+  privateKey,
+  reactive({
+    extendedEditorConfig: {
+      mode: "text/plain",
+      readOnly: false,
+      lineWrapping: true,
+      placeholder: `-----BEGIN PRIVATE KEY-----
+Your private key here
+-----END PRIVATE KEY-----`,
+    },
+    linter: null,
+    completer: null,
+    environmentHighlights: true,
+  })
+)
+
+const algorithms: HoppRESTAuthJWT["algorithm"][] = [
+  "HS256",
+  "HS384",
+  "HS512",
+  "RS256",
+  "RS384",
+  "RS512",
+  "ES256",
+  "ES384",
+  "ES512",
+]
```

Comment:
There are active issues with the usage of asymmetric algorithms. Let's keep only the HMAC-based algorithms in the initial iteration.